include/pressio/ops/eigen/ops_level2.hpp

```diff
@@ -55,6 +55,12 @@ namespace pressio{ namespace ops{
  * y = beta * y + alpha*op(A)*x
 */
 
+// Implementation notes.
+// Eigen requires that:
+// - scalar type of A, x and y is the same
+// - alpha and beta types are same as scalar type
+// - scalar can be constructed from double, eg. Scalar(1)
```

Comment:
these notes should not be here, because if they are here it implicitly means they apply to all overloads. We should either rewrite this better to express more clearly that these apply to any overload where we use native eigen operations or we need to write this note for every overload so that we don't generalize.

Reply:
Would you have any suggestions on how to phrase it more clearly ?
Nothing better than _implementation notes_ on _Eigen requirements_ comes to my mind...